### PR TITLE
Move all tests into test folder

### LINKED ISSUE #150: Put all tests under test/
Let's have `test/unit/*` and `test/e2e/*` under the same `test/` folder vs. putting `e2e` in the root.

### PR DESCRIPTION
**Pull request by Eakam1007**
Resolves #150 

- Renamed `test` folder to `unit`
- Added new folder called `test`
- Moved `unit` and `e2e` folders into `test` folder
- Updated references to paths in/out of `test` folder

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -30,9 +30,9 @@ services:
       - '5053:5000'
 
   # Let's Encrypt test server - https://github.com/letsencrypt/pebble
-  # Extra SSL Cert for testing available at: `test/resources/ca.cert.pem`
+  # Extra SSL Cert for testing available at: `test/unit/resources/ca.cert.pem`
```
Comment (humphd) [inline docker-compose.yml:33]: Remove this, we aren't including that file anymore.

File: docker-compose.yml [hunk 2]
```diff
@@ -30,9 +30,9 @@ services:
...
   # To use in code/tests, add to node's certificate chain via env var:
-  # `cross-env NODE_EXTRA_CA_CERTS=./test/resources/ca.cert.pem node ...`
+  # `cross-env NODE_EXTRA_CA_CERTS=./test/unit/resources/ca.cert.pem node ...`
```
Comment (humphd) [inline docker-compose.yml:35]: Remove this too.  Anything about the extra certs in test/ can get deleted.

File: test/unit/dns.server.test.ts [hunk 3]
```diff
@@ -5,7 +5,7 @@ import {
   deleteRecord,
   getChangeStatus,
   RecordType,
-} from '../app/lib/dns.server';
+} from '../../app/lib/dns.server';
```
Comment (humphd) [inline test/unit/dns.server.test.ts:8]: You should be able to use `~/lib/dns.server` to avoid this problem.

**Review by humphd — CHANGES_REQUESTED** [review]

**Review by humphd — APPROVED** [review]

**Review by cychu42 — APPROVED** [review]
LGTM!

**Pull request merged**